pkg/controller/s3/s3bucketpolicy/s3bucketpolicy.go

```diff
@@ -0,0 +1,222 @@
+	cr, ok := mgd.(*bucketv1alpha3.S3BucketPolicy)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	resp, err := e.client.GetBucketPolicyRequest(&awss3.GetBucketPolicyInput{
+		Bucket: aws.String(cr.Spec.BucketName),
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(s3.IsErrorNotFound, err), errGet)
+	}
+
+	policyData, err := e.formatBucketPolicy(cr)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(s3.IsErrorNotFound, err), errGet)
+	}
+
+	cr.SetConditions(runtimev1alpha1.Available())
+
+	if !cmp.Equal(*policyData, *resp.Policy) {
+		return managed.ExternalObservation{
+			ResourceExists:   true,
+			ResourceUpToDate: false,
+		}, nil
+	}
+
+	// If our version and the external version are the same, we return ResourceUpToDate: true
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: true,
+	}, nil
+}
+
+//formatBucketPolicy parses and formats the bucket.Spec.BucketPolicy struct
+func (e *external) formatBucketPolicy(original *bucketv1alpha3.S3BucketPolicy) (*string, error) {
+	c := original.DeepCopy()
+	iamUsername := c.Spec.UserName
+	accountID, err := e.iamclient.GetAccountID()
+	if err != nil {
+		return nil, err
+	}
+	statements := c.Spec.PolicyBody.PolicyStatement
+	newStatements := make([]bucketv1alpha3.S3BucketPolicyStatement, 0)
+	for _, statement := range statements {
+		if statement.ApplyToIAMUser {
+			statement.Principal.AWSPrincipal = append(statement.Principal.AWSPrincipal, fmt.Sprintf("arn:aws:iam::%s:user/%s", accountID, iamUsername))
+		}
+		statement.ApplyToIAMUser = false
+		updatedPaths := make([]string, 0)
+		for _, v := range statement.ResourcePath {
+			formatted := fmt.Sprintf("arn:aws:s3:::%s", v)
+			updatedPaths = append(updatedPaths, formatted)
+		}
+		statement.ResourcePath = updatedPaths
+		newStatements = append(newStatements, statement)
+	}
+	c.Spec.PolicyBody.PolicyStatement = newStatements
+	body, err := c.Spec.PolicyBody.Serialize()
+	if err != nil {
+		return nil, err
+	}
+	byteData, err := json.Marshal(body)
+	if err != nil {
+		return nil, err
+	}
+	str := string(byteData)
+	return &str, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*bucketv1alpha3.S3BucketPolicy)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.SetConditions(runtimev1alpha1.Creating())
+
+	policyData, err := e.formatBucketPolicy(cr)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errAttach)
+	}
+
+	policyString := *policyData
+	_, err = e.client.PutBucketPolicyRequest(&awss3.PutBucketPolicyInput{Bucket: aws.String(cr.Spec.BucketName), Policy: aws.String(policyString)}).Send(context.TODO())
+	return managed.ExternalCreation{}, errors.Wrap(err, errAttach)
+}
+
+// Update does not do anything
```

Comment:
If `Update` truly does not do anything (i.e. this resource is immutable) then we can return immediately.